docs/common/DocsPageTemplate/SideNav/index.vue

```diff
@@ -77,22 +79,54 @@
         return toc;
       },
     },
+
     watch: {
       filterText(newValue) {
         if (window) {
-          window.sessionStorage.setItem('nav-filter', newValue);
+          //Clear the filter query when filtertext is empty
+          if (!newValue) {
+            this.$router.replace({ path: this.$route.path, query: {} });
+          } else {
+            //else ,update the filter query param
+            this.$router.push({
+              path: this.$route.path,
+              query: { ...this.$route.query, filter: newValue },
+            });
+          }
+          this.debouncedUpdateQuery(newValue);
```

Comment:
As you took another approach with the popstate listener, lets just remove this `debouncedUpdateQuery` call. As there we just do the same logic of pushing the query.

```suggestion
```